internal/static/metrics/wal/wal_test.go

```diff
@@ -116,8 +118,120 @@ func TestStorage(t *testing.T) {
+
+			expectedSamples = append(expectedSamples, commit1Samples...)
+			expectedHistograms = append(expectedHistograms, commit1Histograms...)
+			expectedFloatHistograms = append(expectedFloatHistograms, commit1FloatHistograms...)
+
+			actualSamples := collector.samples
+			require.Equal(t, expectedSamples, actualSamples)
+
+			actualHistograms := collector.histograms
+			require.Equal(t, expectedHistograms, actualHistograms)
+
+			actualFloatHistograms := collector.floatHistograms
+			require.Equal(t, expectedFloatHistograms, actualFloatHistograms)
+
+			// TODO: Verify exemplars were written
+			// expectedExemplars := payload.ExpectedExemplars()
+			// actualExemplars := collector.exemplars
+			// sort.Sort(byRefExemplar(actualExemplars))
+			// require.Equal(t, expectedExemplars, actualExemplars)
+
+			// Verify metadata was written.
+			// The amount of metadata is always the same after every Commit(), because only the first Commit() added metadata.
+			// TODO: Actually the metadata should be len(payload). Fix the issue causing it to be 2*len(payload).
+			require.Len(t, collector.metadata, 2*len(payload), "Expected metadata for each series")
+
+			// Build a map from Ref to series name for type verification
+			refToName := make(map[chunks.HeadSeriesRef]string)
+			for _, s := range collector.series {
+				refToName[s.Ref] = s.Labels.Get("__name__")
+			}
+
+			// Expected types based on series name
+			expectedTypes := map[string]uint8{
+				"float_series":      record.GetMetricType("gauge"),
+				"integer histogram": record.GetMetricType("histogram"),
+				"float histogram":   record.GetMetricType("histogram"),
+				"integer NHCB":      record.GetMetricType("histogram"),
+				"float NHCB":        record.GetMetricType("histogram"),
+			}
+
+			for _, meta := range collector.metadata {
+				name := refToName[meta.Ref]
+				expectedType, ok := expectedTypes[name]
+				require.True(t, ok, "Unknown series name: %s", name)
+				require.Equal(t, expectedType, meta.Type, "Wrong metric type for %s", name)
+				require.Equal(t, "bytes", meta.Unit)
+				require.NotEmpty(t, meta.Help)
+			}
+
+			require.NoError(t, onNotify.Wait(time.Minute), "Expected Notify to be called")
+		})
+	}
+
+	expectedMetrics := `# HELP prometheus_remote_write_wal_metadata_appended_total Total number of metadata entries appended to the WAL
+# TYPE prometheus_remote_write_wal_metadata_appended_total counter
+prometheus_remote_write_wal_metadata_appended_total 10
+# HELP prometheus_remote_write_wal_samples_appended_total Total number of samples appended to the WAL
+# TYPE prometheus_remote_write_wal_samples_appended_total counter
+prometheus_remote_write_wal_samples_appended_total 40
+`
+
+	err = testutil.GatherAndCompare(
+		reg,
+		strings.NewReader(expectedMetrics),
+		"prometheus_remote_write_wal_metadata_appended_total",
+		"prometheus_remote_write_wal_samples_appended_total",
+	)
+	require.NoError(t, err, "expected prometheus registry to be empty")
```

Comment:
```suggestion
	require.NoError(t, err)
```

The assert error is a bit misleading and I don't think we need an extra message but depends on how detailed the err is.

Reply:
I agree. This got autogenerated and I removed several instances of it, but missed tis one.